Firmware/cardreader.cpp

```diff
@@ -69,12 +70,15 @@ char *createFilename(char *buffer,const dir_t &p) //buffer>12characters
 +*/
 
 void CardReader::lsDive(const char *prepend, SdFile parent, const char * const match/*=NULL*/) {
+	static uint8_t recursionCnt = 0;
```

Comment:
This will get initialized only once - the first time you enter this function.
I'm afraid there will be scenarios when this variable will be left in non-zero state thus breaking successive attempts to perform lsDive()

Reply:
Updated the PR to use a RAII principle for the incrementer. There shouldn't exist any case where recursionCnt is non-0 after the last return unless the stack got corrupted